Zend/zend_object_handlers.c

```diff
@@ -175,6 +176,134 @@ ZEND_API HashTable *zend_std_get_debug_info(zend_object *object, int *is_temp) /
 }
 /* }}} */
 
+
+static int zend_std_call_op_handler(zend_uchar opcode, zval *result, zval *op1, zval *op2) /* {{{ */
+{
+	zend_class_entry *ce;
+	zend_object *zobj;
+	if(Z_TYPE_P(op1) == IS_OBJECT) {
+		zobj = Z_OBJ_P(op1);
+		ce = Z_OBJCE_P(op1);
+	} else if(Z_TYPE_P(op2) == IS_OBJECT) {
+		zobj = Z_OBJ_P(op2);
+		ce = Z_OBJCE_P(op2);
+	} else {
+		return FAILURE;
+	}
+
+	zend_class_entry *orig_fake_scope = EG(fake_scope);
+	zend_fcall_info fci;
+	zend_fcall_info_cache fcic;
+	zval params[2] = {*op1, *op2};
+
+	EG(fake_scope) = NULL;
+
+	/* op handlers like __add are called with two operands op1, op2 */
+	fci.size = sizeof(fci);
+	fci.retval = result;
+	fci.param_count = 2;
+	fci.params = params;
+	fci.no_separation = 1;
+	//ZVAL_UNDEF(&fci.function_name); /* Unused */
+
+	do
+	{
+		fci.object = zobj;
+
+		/* Determine what handler should be used, based on opcode */
+		switch (opcode)
+		{
+		case ZEND_ADD:
+			fcic.function_handler = ce->__add;
+			ZVAL_STRING(&fci.function_name, ZEND_ADD_FUNC_NAME);
```

Comment:
Is this actually needed? If the `ce->__add` etc are properly pre-initialized, then it should not be necessary to set the function name here.

Reply:
The idea behind this, was to save the name of magic function called, so it can be for the usage notice below.
Maybe I should move this to the `zend_error(E_NOTICE, ..)` line, so the var is only created if really needed.